index.bs

```diff
@@ -691,21 +691,124 @@ An {{MLOperand}} represents an intermediary graph being constructed as a result
+    : <dfn>\[[activation]]</dfn> of type {{MLActivation}}
+    ::
+        The {{MLOperand}}'s associated activation function object
+    : <dfn>\[[builder]]</dfn> of type {{MLGraphBuilder}}
+    ::
+        The {{MLOperand}}'s associated builder object
+</dl>
+
+
+<div algorithm>
+    The <dfn for="MLOperand">rank</dfn> of an {{MLOperand}} |operand| is the value returned by the following steps:
+    1. Return the size of |operand|.{{MLOperand/[[dimensions]]}}.
+</div>
+
+Since the {{MLOperand/[[builder]]}} object is bound by the {{MLGraphBuilder/constructor()}} constructor to an {{MLContext}} object, an {{MLOperand}} is also always bound to the same {{MLContext}} object.
+
+The guidelines in [[#security-new-ops]] also apply for operands.
+
+#### The {{MLOperand}} constructor #### {#api-mloperand-constructor}
+The [=new=] {{MLOperand}} constructor steps are:
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, throw a "{{SecurityError}}" {{DOMException}} and abort these steps.
+1. Let |builder| be the first argument.
+1. If |builder| is not an instance of {{MLGraphBuilder}}, throw a "{{TypeError}}" and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[builder]]}} to |builder|.
+1. Let |kind| be the second argument.
+1. If |kind| is not an {{MLOperandKind}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[kind]]}} to |kind|.
+1. Let |type| be the third argument.
+1. If |type| is not an {{MLOperandType}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[type]]}} to |type|.
+1. If |kind| is `"constant"`:
+    1. Let |name| be the fourth argument.
+    1. If |name| is not a [=string=], throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[name]]}} to |name|.
+    1. Let |desc| be the fifth argument.
+    1. If |desc| is not an {{MLOperandDescriptor}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |desc|'s {{MLOperandDescriptor/type}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[dimensions]]}} to |desc|.{{MLOperandDescriptor/dimensions}}.
+    1. Let |bufferView| be the sixth argument.
+    1. If |bufferView| is not an {{MLBufferView}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |bufferView|'s [=element type=], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not equal with |bufferView|.\[[ByteLength]], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[tensorData]]}} to |bufferView|.
```

Comment:
Should `MLOperand` have corresponding platform object? Should the `kind`, `type`, `dimensions` and `values` be used to create the platform object?